Gems/EMotionFX/Code/Source/Editor/Plugins/Ragdoll/RagdollColliderTranslationManipulators.cpp

```diff
@@ -0,0 +1,193 @@
+                BeginEditing(action.m_start.m_localPosition, action.m_current.m_localOffset);
+            });
+
+        m_translationManipulators.InstallSurfaceManipulatorMouseDownCallback(
+            [this](const AzToolsFramework::SurfaceManipulator::Action& action)
+            {
+                BeginEditing(action.m_start.m_localPosition, action.m_current.m_localOffset);
+            });
+
+        // mouse move callbacks
+        m_translationManipulators.InstallLinearManipulatorMouseMoveCallback(
+            [this](const AzToolsFramework::LinearManipulator::Action& action)
+            {
+                OnManipulatorMoved(action.m_start.m_localPosition, action.m_current.m_localPositionOffset);
+            });
+
+        m_translationManipulators.InstallPlanarManipulatorMouseMoveCallback(
+            [this](const AzToolsFramework::PlanarManipulator::Action& action)
+            {
+                OnManipulatorMoved(action.m_start.m_localPosition, action.m_current.m_localOffset);
+            });
+
+        m_translationManipulators.InstallSurfaceManipulatorMouseMoveCallback(
+            [this](const AzToolsFramework::SurfaceManipulator::Action& action)
+            {
+                OnManipulatorMoved(action.m_start.m_localPosition, action.m_current.m_localOffset);
+            });
+
+        // mouse up callbacks
+        m_translationManipulators.InstallLinearManipulatorMouseUpCallback(
+            [this](const AzToolsFramework::LinearManipulator::Action& action)
+            {
+                FinishEditing(action.m_start.m_localPosition, action.m_current.m_localPositionOffset);
+            });
+
+        m_translationManipulators.InstallPlanarManipulatorMouseUpCallback(
+            [this](const AzToolsFramework::PlanarManipulator::Action& action)
+            {
+                FinishEditing(action.m_start.m_localPosition, action.m_current.m_localOffset);
+            });
+
+        m_translationManipulators.InstallSurfaceManipulatorMouseUpCallback(
+            [this](const AzToolsFramework::SurfaceManipulator::Action& action)
+            {
+                FinishEditing(action.m_start.m_localPosition, action.m_current.m_localOffset);
+            });
+    }
+
+    void RagdollColliderTranslationManipulators::Refresh()
+    {
+        if (!m_ragdollManipulatorData.m_valid || !m_ragdollManipulatorData.m_colliderNodeConfiguration ||
+            m_ragdollManipulatorData.m_colliderNodeConfiguration->m_shapes.empty())
+        {
+            return;
+        }
+
+        m_translationManipulators.SetLocalPosition(m_ragdollManipulatorData.m_colliderNodeConfiguration->m_shapes[0].first->m_position);
+    }
+
+    void RagdollColliderTranslationManipulators::Teardown()
+    {
+        m_translationManipulators.Unregister();
+    }
+
+    void RagdollColliderTranslationManipulators::ResetValues()
+    {
+        m_ragdollManipulatorData.m_colliderNodeConfiguration->m_shapes[0].first->m_position = AZ::Vector3::CreateZero();
```

Comment:
should this check that `m_shapes` is not empty like in the other functions?

Reply:
Yeah, will add.